enterprise/server/remote_execution/containers/podman/podman.go

```diff
@@ -443,13 +457,87 @@ func (c *podmanCommandContainer) PullImage(ctx context.Context, creds container.
 
 	ps.mu.Lock()
 	defer ps.mu.Unlock()
+	if *imageStreamingEnabled {
+		if err := c.prepareToStreamImage(ctx); err != nil {
+			return err
+		}
+	}
 	if err := c.pullImage(ctx, creds); err != nil {
 		return err
 	}
 	ps.pulled = true
 	return nil
 }
 
+func (c *podmanCommandContainer) prepareToStreamImage(ctx context.Context) error {
+	ctx, err := prefix.AttachUserPrefixToContext(ctx, c.env)
+	if err != nil {
+		return err
+	}
+	resp, err := c.env.GetSociArtifactStoreClient().GetArtifacts(ctx, &socipb.GetArtifactsRequest{Image: c.image})
```

Comment:
you are not passing platform or credentials

Reply:
Argh, whoops, fixed.